In [0]:
import numpy as np # Numpy module for Linear Algebra
import pandas as pd # Pandas module for Data Manipulation
import matplotlib.pyplot as plt # Module for Plotting
from sklearn.model_selection import train_test_split #Splitting data into test and train sets
from sklearn.metrics import r2_score #Module for calculating r_square value
from sklearn.metrics import mean_squared_error #To find the mean squared error value
from sklearn.preprocessing import StandardScaler # Used to scale the data
import math #For computing math functions
import time # Time library to get time
from sklearn.linear_model import LinearRegression # Linear Regression from Sklearn Library to solve Linear Equations.
from sklearn.metrics import r2_score # For checking R Square
from IPython.display import Markdown as md #For displaying results in md
import seaborn as sns # Another module for plotting
from scipy import stats # Module For removing outliers in this program.
%matplotlib inline

In [0]:
# Import csv file using pandas
bags = pd.read_csv ('Data_miniproject.csv');
# Remove rows with cost feature 0  
bags = bags[bags.Cost != 0]


# Eliminate outliers using z-scores from Scipy library.
z_scores = stats.zscore(bags)
abs_z_scores = np.abs(z_scores)
filtered_entries = (abs_z_scores < 3).all(axis=1)
bags = bags[filtered_entries]


# Taking dependent variables as X and independent variable(COST) as y
y = bags.pop('Cost');
X = bags.copy()

# Scaling the Data for faster computations and easy understanding of machine
column_names = X.columns
scaler = StandardScaler()
X = pd.DataFrame(scaler.fit_transform(X),columns=column_names)

# Splitting data into test and train sets 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Creating an object for Linear Regression
multi_reg = LinearRegression();
multi_reg.fit(X_train, y_train)
print("Co_effecient = ",multi_reg.coef_)
print("Intercept = ",multi_reg.intercept_)

Co_effecient =  [ 830.24455568 -338.62302295 -390.9992857   152.02921227   95.58715217]
Intercept =  375.1229399929727


In [0]:
# Checking Predictions
y_pred = multi_reg.predict(X_test)

# Calculating Errors and Score
mse = mean_squared_error(multi_reg.predict(X_train),y_train)
rmse = math.sqrt(mean_squared_error(multi_reg.predict(X),y))
rse = (rmse**2)*X.shape[0];
rse /= X.shape[0]-2
r_square = r2_score(y_test,y_pred);

# Displaying Errors and Score

print('{:>10} {:>20} {:^35} {:^10} '.format("MSE","RMSE","RSE","R SQUARE"))

print('{:<20}| {:>14}  | {:^22}| {:^22} '.format(mse,rmse,rse,r_square))


------------------------------------------------------------------------------------------
       MSE                 RMSE                 RSE                  R SQUARE  
------------------------------------------------------------------------------------------
9839.820568747116   | 98.57942236920306  |   9844.933920065938   |   0.9069783944138913   
------------------------------------------------------------------------------------------


### Conclusion 
<br />
As we can see the RSquare Score is 0.907. This modal fits 90.7% which means that the current modal is a good fit.